tests/contest/contest/src/tests/cgroups/relative_network.rs

```diff
@@ -0,0 +1,147 @@
+use std::fs;
+use std::path::{Path, PathBuf};
+
+use anyhow::{bail, Context, Result};
+use oci_spec::runtime::{
+    LinuxBuilder, LinuxInterfacePriorityBuilder, LinuxNetworkBuilder, LinuxResourcesBuilder, Spec,
+    SpecBuilder,
+};
+use pnet_datalink::interfaces;
+use test_framework::{test_result, ConditionalTest, TestGroup, TestResult};
+
+use crate::utils::test_outside_container;
+use crate::utils::test_utils::{check_container_created, CGROUP_ROOT};
+
+fn create_spec(cgroup_name: &str, class_id: u32, prio: u32, if_name: &str) -> Result<Spec> {
+    // Create the Linux Spec
+    let linux_spec = LinuxBuilder::default()
+        .cgroups_path(Path::new("testdir/runtime-test/container").join(cgroup_name))
+        .resources(
+            LinuxResourcesBuilder::default()
+                .network(
+                    LinuxNetworkBuilder::default()
+                        .class_id(class_id)
+                        .priorities(vec![LinuxInterfacePriorityBuilder::default()
+                            .name(if_name)
+                            .priority(prio)
+                            .build()
+                            .context("failed to build network interface priority spec")?])
+                        .build()
+                        .context("failed to build network spec")?,
+                )
+                .build()
+                .context("failed to build resource spec")?,
+        )
+        .build()
+        .context("failed to build linux spec")?;
+
+    // Create the top level Spec
+    let spec = SpecBuilder::default()
+        .linux(linux_spec)
+        .build()
+        .context("failed to build spec")?;
+
+    Ok(spec)
+}
+
+// Gets the loopback interface if it exists
+fn get_loopback_interface() -> Option<String> {
+    let interfaces = interfaces();
+    let lo_if_name = interfaces.first().map(|iface| &iface.name)?;
+
+    Some(lo_if_name.to_string())
+}
+
+fn test_relative_network_cgroups() -> TestResult {
+    let cgroup_name = "test_relative_network_cgroups";
+
+    let id = 255;
+    let prio = 10;
+    let if_name = "lo";
+    let spec = test_result!(create_spec(cgroup_name, id, prio, if_name));
+
+    let test_result = test_outside_container(spec.clone(), &|data| {
+        test_result!(check_container_created(&data));
+        test_result!(validate_network(cgroup_name, &spec));
+        TestResult::Passed
+    });
+    if let TestResult::Failed(_) = test_result {
+        return test_result;
+    }
+
+    TestResult::Passed
+}
+
+/// validates the Network structure parsed from /sys/fs/cgroup/net_cls,net_prio with the spec
+fn validate_network(cgroup_name: &str, spec: &Spec) -> Result<()> {
+    let cgroup_path = PathBuf::from(CGROUP_ROOT)
+        .join("net_cls,net_prio/runtime-test")
+        .join(cgroup_name);
```

Comment:
Is it ok to hard-code this here? I think in the `can_run` below you mention the cgroup can be at a couple of different  points.

Reply:
Address multiple network cgroup mount points.
Changed where to get paths for `net_cls.classid` and `net_prio.ifpriomap`.
[ca7ab449](https://github.com/youki-dev/youki/pull/2986/commits/ca7ab44933730c0591270e1dd50862db5dc38514)